tests/python/states/states_test.py

```diff
@@ -331,6 +332,90 @@ def test_indices_and_strings_roundtrip_spinful(norb: int, nelec: tuple[int, int]
     np.testing.assert_array_equal(indices_again, indices)
 
 
+def _empirical_distribution(bts_matrix, norb, nelec):
+    indices = np.zeros(bts_matrix.shape[0], dtype=int)
+    for i, bts in enumerate(bts_matrix):
+        string = np.array2string(bts, separator="")[1:-1]
+        index = ffsim.strings_to_indices([string], norb, nelec)[0]
+        indices[i] = index
+
+    unique_indices, counts = np.unique(indices, return_counts=True)
+
+    if isinstance(nelec, tuple):
+        probabilities = np.zeros(math.comb(norb, nelec[0]) * math.comb(norb, nelec[1]))
+    else:
+        probabilities = np.zeros(math.comb(norb, nelec))
+
+    probabilities[unique_indices] = counts
+
+    probabilities /= np.sum(probabilities)
+
+    return probabilities
+
+
+@pytest.mark.parametrize("norb, nelec", ffsim.testing.generate_norb_nelec(range(1, 5)))
+def test_slater_sampler(norb: int, nelec: tuple[int, int]):
+    """Test Slater determinant sampler."""
+    rng = np.random.default_rng(1234)
+
+    n_samples = 3000
+
+    n_a, n_b = nelec
+
+    rotation_a = ffsim.random.random_unitary(norb, seed=rng)
+    rotation_b = ffsim.random.random_unitary(norb, seed=rng)
+
+    vecs_a = rotation_a[:, :n_a]
+    vecs_b = rotation_b[:, :n_b]
+
+    rdm_a = vecs_a @ np.conjugate(vecs_a.T)
+    rdm_b = vecs_b @ np.conjugate(vecs_b.T)
+
+    test_distribution = (
+        np.absolute(
```

Comment:
```suggestion
        np.abs(
```
Maybe this will make the formatting less aggressive?